Optimize CLS - add to dynamic content section (#5966)
GoogleChrome/web.dev: src/site/content/en/blog/optimize-cls/index.md

If you need to display these types of UI affordances, reserve sufficient space in the viewport for it in advance (for example, using a placeholder or skeleton UI) so that when it loads, it does not cause content in the page to surprisingly shift around.

> In some cases adding content dynamically is important part of user experience. For example, when loading more products to a list of items or when updating live feed content. There are several ways to avoid unexpected layout shifts in those cases:

Author: I'm not a native speaker, so mainly confirming - isn't it supposed to be ".. is **an** important part of **the** user experience .."?

Reviewer: Good catch, it is "an important". No need for "the" next to user experience because it's user experience in a general sense.

If you need to display these types of UI affordances, reserve sufficient space in the viewport for it in advance (for example, using a placeholder or skeleton UI) so that when it loads, it does not cause content in the page to surprisingly shift around.

> In some cases adding content dynamically is important part of user experience. For example, when loading more products to a list of items or when updating live feed content. There are several ways to avoid unexpected layout shifts in those cases:

Reviewer: ```suggestion
In some cases adding content dynamically is an important part of user experience. For example, when loading more products to a list of items or when updating live feed content. There are several ways to avoid unexpected layout shifts in those cases:
```

> In some cases adding content dynamically is an integral part of the page. For example, when loading more products to a list of items or when updating live feed content. In those cases you can avoid unexpected layout shifts by:
> 
> *   Replacing the old content with the new content within a fixed size container or by using a carousel and removing the old content after the transition. Remember to disable any links/controls until the transition has completed in order to prevent accidental clicks/taps while the new content is coming in.
> *   Having the user initiate the load of new content such that they will be expecting the shift (e.g. with a "load more" or "refresh" button). It’s recommended to prefetch the content before the user interaction so that it will show up immediately after it. As a reminder, layout shifts that don't occur within 500ms of user input will be counted towards CLS.
> *   Seamlessly load the content offscreen and overlay a notice to the user that it’s available (e.g. with a "scroll up" button).
> 

Reviewer: ```suggestion
In some cases adding content dynamically is important part of user experience. For example, when loading more products to a list of items or when updating live feed content. There are several ways to avoid unexpected layout shifts in those cases:

*   Replace the old content with the new content within a fixed size container or use a carousel and remove the old content after the transition. Remember to disable any links and controls until the transition has completed to prevent accidental clicks or taps while the new content is coming in.
*   Have the user initiate the load of new content, so they are not surprised by the shift (for example with a "Load more" or "Refresh" button). It’s recommended to prefetch the content before the user interaction so that it shows up immediately. As a reminder, layout shifts that occur within 500&nbsp;ms of user input are not counted towards CLS.
*   Seamlessly load the content offscreen and overlay a notice to the user that it’s available (for example, with a "Scroll up" button).

```

<figure class="w-figure">
  {% Img src="image/OcYv93SYnIg1kfTihK6xqRDebvB2/B6oqm3fEXtby7qQadk4D.png", alt="Examples of dynamic content loading without causing unexpected layout shifts from Twitter and the Chloé website", width="579", height="460" %}
  <figcaption class="w-figcaption">

>     Examples of dynamic content loading without causing unexpected layout shifts. Left: "Load More" example on the Chloé website. Check out the related [article](https://medium.com/ynap-tech/how-to-optimize-for-cls-when-having-to-load-more-content-3f60f0cf561c) from the YNAP tech team. Right: Live feed content loading on Twitter.

Reviewer: ```suggestion
    Examples of dynamic content loading without causing unexpected layout shifts. Left: "Load More" example on Chloé website. Check out how the YNAP team [optimized for CLS when loading more content](https://medium.com/ynap-tech/how-to-optimize-for-cls-when-having-to-load-more-content-3f60f0cf561c). Right: Live feed content loading on Twitter.
```

<figure class="w-figure">
  {% Img src="image/OcYv93SYnIg1kfTihK6xqRDebvB2/B6oqm3fEXtby7qQadk4D.png", alt="Examples of dynamic content loading without causing unexpected layout shifts from Twitter and the Chloé website", width="579", height="460" %}
  <figcaption class="w-figcaption">

>     Examples of dynamic content loading without causing unexpected layout shifts. Left: "Load More" example on Chloé website. Check out how the YNAP team [optimized for CLS when loading more content](https://medium.com/ynap-tech/how-to-optimize-for-cls-when-having-to-load-more-content-3f60f0cf561c). Right: Live feed content loading on Twitter.

Author: Again, not a native speaker, but "on Chloe website" seems off, so just making sure, cause I would never say something like "cool doodle on Google website". If the intention was to make it shorter then maybe we can go with - Left: "Load More" example, Chloé website. In this case I would also change the right description to - Live feed content loading, Twitter.

Reviewer: There isn't a strict grammar rule, but "on website" is a very idiomatic ("at" or "in" website is rarely appropriate) and no need for definite article "the" since Chloe is a proper noun
